chromadb/utils/data_loaders.py

```diff
@@ -22,3 +23,123 @@ def _load_image(self, uri: Optional[URI]) -> Optional[Image]:
     def __call__(self, uris: Sequence[Optional[URI]]) -> List[Optional[Image]]:
         with ThreadPoolExecutor(max_workers=self._max_workers) as executor:
             return list(executor.map(self._load_image, uris))
+
+# --------------------------- A lightweight util ---------------------------
+
+# TODO: Placing this here, since data loaders are a common use case for this function.
+#   But it should probably be moved to a more general-purpose module.
+#   See: https://github.com/chroma-core/chroma/issues/1606
+from functools import partial
+from concurrent.futures import ThreadPoolExecutor
+
+def vectorize(func, iterable=None, *, max_workers : int = 1):
```

Comment:
Regarding the `vectorize` name -- I see what you mean (since... chroma is a vector DB?).

What names would you suggest?

ChatGPT tells me the process is usually called: mapping, vectorize (thus my choice), batch processing, broadcasting, and apply.

If no better suggestions, I'd propose `mapper`. 
Because really I'm just doing a `map` (serial of parallel version) -- the only differences being:
* It's set up (and meant to be used) to be a *factory* -- i.e. fixing the function, and then applying it to arrays of inputs. So `mapper(func)` says "make a mapper using this func". 
* It has a "parallelize this" option, through the `max_workers` argument.

Regarding this `max_workers` argument: I put it there because I wanted to stay aligned with the only example of a `DataLoader` I could find: Namely `ImageLoader`.
Note though that I did **not** have the default be `multiprocessing.cpu_count()` as it is in `ImageLoader`. 

I'm conflicted here because:
* On one hand, I like everything to be as aligned as possible, so maybe we should have `multiprocessing.cpu_count()` be the default of all `max_workers` of our data loaders?
* On the other hand, I made a few tests and the outcome (with local file io) was always that the serial version was faster. That said, one could make the argument that (1) it's only when the numbers become big that the efficiency is needed, and appears and (2) that remote file io would probably yield different results, favoring parallelism. 

What do you think?

Reply:
I think we don't have to go to typevars, we often use `Any/None` especially for `Callables`.  e.g. `Callable[[Request], Any]` should be good enough.